### PR TITLE
APPS-1482: adds more models

### PR DESCRIPTION
**Pull request by aaronepinto**
[APPS-1482](https://dataloop.atlassian.net/jira/software/c/projects/DAT/boards/16?assignee=712020%3A4bb14cfa-051b-43fb-a69e-833497de40d5&selectedIssue=APPS-1482)

This pull request introduces significant changes to the Databricks model adapters. The changes include the following:

- Expanded the `README.md` with detailed steps for configuring the Databricks chat completion and text embeddings model adapter
- Implemented a new model adapter class for text embeddings
- Generalized the model adapter class for chat completions.
- Removed the dbrx-instruct specific model 

Some items to discuss before merging:

- [ ] Which other models should we add support for before merging the changes?
- [ ] Does the generalized approach work or would we still prefer model specific adapters?
- [ ] How should we integrate testing?

[APPS-1482]: https://dataloop.atlassian.net/browse/APPS-1482?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: adapters/meta_llama_3/dataloop.json [hunk 1]
```diff
@@ -0,0 +1,139 @@
+{
+  "name": "databricks-meta-llama-3",
+  "displayName": "Meta Llama 3",
+  "version": "0.0.6",
+  "scope": "public",
+  "description": "The Llama 3 models are large language models (LLMs) developed by Meta AI. In order to use this model, it is necessary to have a Databricks serving endpoint and add a Databricks Token as a secret. For more information, consult the README files in the dataloop-ai-apps repo.",
+  "attributes": {
+    "Provider": "Databricks",
```
Comment (RoniDataloopAi) [inline adapters/meta_llama_3/dataloop.json:8]: provider meta

File: adapters/meta_llama_3/dataloop.json [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      }
+    ],
+    "modules": [
+      {
+        "name": "meta-llama-3-module",
+        "entryPoint": "adapters/base_chat_completion.py",
+        "className": "ModelAdapter",
+        "computeConfig": "meta-llama-3-deploy",
+        "description": "dbrx instruct module",
+        "integrations": [
```
Comment (RoniDataloopAi) [inline adapters/meta_llama_3/dataloop.json:47]: change the description

File: adapters/text_embeddings/text_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import openai
+from openai import NOT_GIVEN
```
Comment (RoniDataloopAi) [inline adapters/text_embeddings/text_embeddings.py:2]: this one shouldn't be removed? the all folder?

File: adapters/base_embeddings.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    def load(self, local_path, **kwargs):
+        """Load configuration for Databricks adapter"""
+        api_key = os.environ.get("DATABRICKS_API_KEY")
+        base_url = self.configuration.get("base_url")
+    
+        if not api_key:
+            raise ValueError(
+                "Environment variable 'DATABRICKS_API_KEY' is not set. "
+                "Please set it to proceed."
+            )
```
Comment (RoniDataloopAi) [inline adapters/base_embeddings.py:20]: the secret name can be any name, not necessarily 'DATABRICKS_API_KEY'

File: adapters/base_chat_completion.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    def load(self, local_path, **kwargs):
+        api_key = os.environ.get("DATABRICKS_API_KEY", None)
+        if api_key is None:
+            raise ValueError("Missing API key: DATABRICKS_API_KEY")
+
+        self.client = OpenAI(
+            api_key=api_key,
+            base_url=self.model_entity.configuration.get("base_url")
+        )
+
```
Comment (RoniDataloopAi) [inline adapters/base_chat_completion.py:21]: add here also the url check? like in the base embedder

File: adapters/base_chat_completion.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    def prepare_item_func(self, item: dl.Item):
+        prompt_item = dl.PromptItem.from_item(item)
+        return prompt_item
+
+    def stream_response(self, messages):
+        stream = self.configuration.get("stream", True)
+        response = self.client.chat.completions.create(
+            messages=messages,
+            max_tokens=self.configuration.get("max_tokens", openai.NOT_GIVEN),
```
Comment (RoniDataloopAi) [inline adapters/base_chat_completion.py:30]: you can import directly NOT_GIVEN

File: adapters/text_embeddings/dataloop.json [hunk 7]
```diff
@@ -0,0 +1,117 @@
+{
+  "name": "databricks-text-embeddings",
```
[the remark this answers is not in the document]
Comment (aaronepinto) [reply adapters/text_embeddings/dataloop.json:2]: Done, removed ✅

File: adapters/text_embeddings/text_embeddings.py (hunk 3 above)
Comment (aaronepinto) [reply adapters/text_embeddings/text_embeddings.py:2]: Done ✅

File: adapters/base_embeddings.py (hunk 4 above)
Comment (aaronepinto) [reply adapters/base_embeddings.py:20]: @RoniDataloopAi the previous dbrx code i.e., the base_chat_completion.py does it this way -- do you have a recommended alternative?

File: adapters/meta_llama_3/dataloop.json (hunk 1 above)
Comment (aaronepinto) [reply adapters/meta_llama_3/dataloop.json:8]: Done ✅

File: adapters/base_chat_completion.py (hunk 5 above)
Comment (aaronepinto) [reply adapters/base_chat_completion.py:21]: Done ✅

File: adapters/base_chat_completion.py (hunk 6 above)
Comment (aaronepinto) [reply adapters/base_chat_completion.py:30]: Done ✅

**Pull request merged**